api/JsonBuilder.h

```diff
@@ -0,0 +1,133 @@
+#pragma once
+#include <cstring>
+
+/**
+ * Helper objects used to format a json body. Use code blocks to manage life time of JsonArray and JsonObject to set the
+ * braces inside the json. You have to add elements in order. You cannot add ddproperties to an object higher in
+ * hierarchy until you have closed the current object.
+ * example:
+ * Stringbuilder b;
+ * {
+ *  JsonObject json(b);
+ *  {
+ *      JsonObject foo(b, "foo");
+ *      foo.addProperty("test", 56);
+ *      foo.addProperty("tar", "string");
+ *      // json.addProperty("error",7); this is invalid
+ *      JsonArray bar("bar");
+ *      for (int i = 0; i < 25;++i)
+ *      {
+ *          bar.addElement(i*7);
+ *      }
+ *  }
+ * }
+ * */
+namespace JsonBuilder
+{
+
+template <typename TBuilder>
+class JsonObject
+{
+    TBuilder& _builder;
+
+public:
+    JsonObject(TBuilder& builder, const char* name) : _builder(builder)
+    {
+        assert(name);
+        if (!(_builder.empty() || _builder.endsWidth('[') || _builder.endsWidth('{')))
+        {
+            _builder.append(",");
+        }
+
+        _builder.append("\"");
+        _builder.append(name);
+        _builder.append("\":");
+
+        _builder.append("{");
+    };
+
+    JsonObject(TBuilder& builder) : _builder(builder)
+    {
+        if (!(_builder.empty() || _builder.endsWidth('[') || _builder.endsWidth('{')))
+        {
+            _builder.append(",");
+        }
+
+        _builder.append("{");
+    };
+
+    ~JsonObject() { _builder.append("}"); }
+
+    void addProperty(const char* name, const char* value)
```

Comment:
Consider using generic "Builder" pattern, where method adding properties or sub-builders returns *this (as JsonObjectBuilder) and only final "build()" returns  final constructed object. That would allow to chain things, e.g.:

JsonObjectBuilder builder;
JsonObject obj = builder.addProperty("foo", "bar")
                        .addProperty(...)
                        ...
                        .addArray(...)
                              .addElem(...)
                              ...
                              .build()
                        .build();

Reply:
Considered. The purpose of these objects are to leverage the constructor destructor to manage the braces and create a structure in code that represents the json.
lohmann json would have been better but it entails mem alloc and exceptions.